### ISSUE TITLE
Ionic/angular 7 support 

### ISSUE THREAD

**Issue by sdsoldi** [twinssbc/Ionic2-Calendar#650]
Found: @ionic/angular@7.0.0
node_modules/@ionic/angular
    @ionic/angular@"^7.0.0" from the root project

Could not resolve dependency:
peer @ionic/angular@"^6.1.9" from ionic2-calendar@1.0.1
node_modules/ionic2-calendar
    ionic2-calendar@"^1.0.1" from the root project

**Comment by nashory** [comment]
+1

**Comment by bhandaribhumin** [comment]
Same for me:
```

peer @ionic/angular@"^6.1.9" from ionic2-calendar@1.0.1
npm ERR! node_modules/ionic2-calendar
npm ERR!   ionic2-calendar@"^1.0.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**Comment by nzomedia** [comment]
+1

**Comment by bhandaribhumin** [comment]
Published new package [https://www.npmjs.com/package/ionic2-calendar-16](https://www.npmjs.com/package/ionic2-calendar-16)
Compatible with Angular 16 and Swiper 9
 
WARNING: NOT TESTED

**Comment by Eraldo** [comment]
> Published new package https://www.npmjs.com/package/ionic2-calendar-16 Compatible with Angular 16 and Swiper 9
> 
> WARNING: NOT TESTED

Tried it. The imports didn't work for me.
I also changed them to *-16 and -16/dist/...
Didn't manage to get the imports to work either way.
Looking forward to finding a fix. :)
Thanks @bhandaribhumin for the work you put into this. 👍

**Comment by fitnexus** [comment]
any update on this?

**Comment by twinssbc** [comment]
working on it

**Comment by twinssbc** [comment]
I have published a new version, 2.0.0, feel free to give a try.
One thing to note is that the I'm still working on Swiper 9 integration. So when installing the Swiper dependency, just use 8.4.6 version.

**Comment by narik1989** [comment]
Hi, 

I did install the latest 2.x.x and downgraded my swiper to 8.4.6. I am not getting the proper screen interface in the calendar page.

Please find below the set up information
Ionic:

   Ionic CLI                     : 7.1.1 (C:\Users\kirankumar.s\AppData\Roaming\nvm\v18.16.0\node_modules\@ionic\cli)
   Ionic Framework               : @ionic/angular 7.0.7
   @angular-devkit/build-angular : 16.0.2
   @angular-devkit/schematics    : 16.0.2
   @angular/cli                  : 16.0.2
   @ionic/angular-toolkit        : 9.0.0

Capacitor:

   Capacitor CLI      : 5.0.3
   @capacitor/android : 5.0.3
   @capacitor/core    : 5.0.3
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 1.7.2

System:

   NodeJS : v18.16.0 (C:\Program Files\nodejs\node.exe)
   npm    : 9.5.1
   OS     : Windows 10
   
   Installed ionic calendar version - "ionic2-calendar": "^2.0.0",

![calendar](https://github.com/twinssbc/Ionic2-Calendar/assets/33231617/79e2f8ff-c419-48e6-aab2-66a7694f6469)

Code in Html 

<calendar #myCalendar [timeInterval]="30" [dateFormatter]="calendar.dateFormatter"></calendar>

The import statement in page.ts
import { CalendarComponent } from "ionic2-calendar";

The import in module of the page

import { NgCalendarModule } from 'ionic2-calendar';

Please let me know what i have missed so i could verify it again. Any help would be greatly appreciated.

Regards
Kiran

**Comment by JamesDAdams** [comment]
> I have published a new version, 2.0.0, feel free to give a try. One thing to note is that the I'm still working on Swiper 9 integration. So when installing the Swiper dependency, just use 8.4.6 version.

I try the 2.0.0 version, but he need @angular/common 16.0.0 :(

**Comment by twinssbc** [comment]
I have published version 2.1.0, which supports swiper 10.1.0. Feel free to give a try to see if the issue still exists.
